probe/docker/container.go

```diff
@@ -303,11 +306,20 @@ func (c *container) NetworkInfo(localAddrs []net.IP) report.Sets {
 	// Treat all Docker IPs as local scoped.
 	ipsWithScopes := addScopeToIPs(c.hostID, ipv4s)
 
-	return report.MakeSets().
-		Add(ContainerNetworks, report.MakeStringSet(networks...)).
-		Add(ContainerPorts, c.ports(localAddrs)).
-		Add(ContainerIPs, report.MakeStringSet(ipv4s...)).
-		Add(ContainerIPsWithScopes, report.MakeStringSet(ipsWithScopes...))
+	s := report.MakeSets()
+	if len(networks) > 0 {
+		s = s.Add(ContainerNetworks, report.MakeStringSet(networks...))
+	}
+	if len(c.container.NetworkSettings.Ports) > 0 {
+		s = s.Add(ContainerPorts, c.ports(localAddrs))
+	}
+	if len(ipv4s) > 0 {
+		s = s.Add(ContainerIPs, report.MakeStringSet(ipv4s...))
+	}
+	if len(ipsWithScopes) > 0 {
+		s = s.Add(ContainerIPsWithScopes, report.MakeStringSet(ipsWithScopes...))
+	}
+	return s
```

Comment:
it's kinda tempting to introduce a `Sets.AddMulti` (or s.t. like that) which does the len check and conditional StringSet construction. There are probably a few other places where we could use that.